modules/afmongodb/afmongodb-worker.c

```diff
@@ -65,6 +112,12 @@ _switch_collection(MongoDBDestWorker *self, const gchar *collection)
   if (!self->client)
     return FALSE;
 
+  if (self->bulk_op)
+    {
+      mongoc_bulk_operation_destroy(self->bulk_op);
+      self->bulk_op = NULL;
+    }
```

Comment:
Maybe I misunderstood something, but `_switch_collection()` seems to be broken with bulk sending.

We are currently switching collections before each insertion, but bulk sending requires a `flush()` call in order to send out messages. With the current code, we'd lose the entire batch during collection switch  (when using a templated `collection()`).  We either should trigger a flush manually before switching collection (it can be done in insert()), or we should not support templates while operating in bulk.

I'd go with the first option and document that bulk sending is only efficient if the user sets a constant collection or a template that does not use too many collections within a reasonable time range.

Reply:
upps, i did not notice the switch in the insert, i just saw the one in connect
will try to fix this